DateOfBirthSorter/src/BirthSorter.java

```diff
@@ -0,0 +1,36 @@
+import java.io.BufferedReader;
+import java.io.FileNotFoundException;
+import java.io.FileReader;
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+import java.util.Arrays;
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+public class BirthSorter {
+    public static void main(String[] args) {
+
+        System.out.println(sortBirthDate());
+    }
+
+    public  static List<String> sortBirthDate() {
+        try {BufferedReader bufferedReader = new BufferedReader(new FileReader("data.txt"));
+
+            Stream<String> lines = bufferedReader.lines();
+            DateTimeFormatter dateTimeFormatter = DateTimeFormatter.ofPattern("yyyy-MM-dd");
+
+            List<String> sortedLines = lines.map((l) -> Arrays.asList(l.split(",")))
+                    .sorted((lin1, lin2) -> LocalDate.parse(lin1.get(lin1.size() - 1), dateTimeFormatter)
```

Comment:
This can be simplified if you add class Person to your model, which has 2 fields: String name, LocalDate dateOfBirth. Then you map from a string line to Person, then just sort persons by date and then map resulting persons to strings.

Reply:
I have done using class Person.

https://github.com/Arkarine/Homeworks/pull/10/commits/6488dc2e3a4762df48eec52cb91ec8a395306e21